.github/workflows/swap-azure-web-app-slots.yml

```diff
@@ -69,6 +76,28 @@ on:
         description: >
           ID of the Application Insights resource that the release annotation for the swap should be added to.
           This can e.g. be looked up on the Azure Portal under the given AI resource's Overview page -> JSON View.
+      app-code-repo:
```

Comment:
With this, now we can know the SHA of the commit that was swapped out. The `add-azure-application-insights-release-annotation` action below adds an SHA to the AI release annotation ([here](https://github.com/Lombiq/GitHub-Actions/blob/144cff309e4c280f4b67a64f96c3624e23476b63/.github/actions/add-azure-application-insights-release-annotation/action.yml#L37)), but for swaps, this is the SHA of the commit in the swap repo.

Please change the action so it accepts an options parameter for the SHA. That would default to `'${{ github.sha }}'` and works as before if no parameter is passed, but pass the SHA for the code repo's commit in this workflow.

Reply:
To clarify: I'm talking about adding a new parameter to the `add-azure-application-insights-release-annotation` action (not `swap-azure-web-app-slots`, which is also a workflow, not an action).

So, we can pass an SHA to it optionally. If not passed, it'll just use the current SHA as it does now. But here in this workflow, you can pass an explicit SHA, the SHA of the code repo commit that's being swapped out. Because currently the SHA of the Swap repo commit is recorded in the release annotation (for access control we keep the swap workflows in repos separate from the code repos as you've seen).

This is how it looks like in the Azure Portal:

![image](https://github.com/user-attachments/assets/c5c5c401-73d8-4212-af2b-c5b2a4828c30)

"Commit SHA" there is the commit in the swap repo. That's useless to try to correlate a state of the codebase with a swap. Instead, we need the SHA of the code repo. And this is now easily available.

Reply:
Gotcha, this makes more sense and have implemented it this way now

Reply:
I see the parameter in the add-azure-application-insights-release-annotation action but the swap workflow doesn't use it. Feel free to move the "Add Azure Application Insights Release Annotation" step down BTW, since it'll need to be after the code repo checkout.

Reply:
This isn't addressed.